runtime/src/nonce_keyed_account.rs

```diff
@@ -281,15 +268,9 @@ pub fn authorize_nonce_account(
                 );
                 return Err(InstructionError::MissingRequiredSignature);
             }
-            let new_data = nonce::state::Data::new(
-                *nonce_authority,
-                data.durable_nonce,
-                data.get_lamports_per_signature(),
-            );
-            account.set_state(&Versions::new(
-                State::Initialized(new_data),
-                separate_domains,
-            ))
+            // Safe to unwrap here because state is known to be Initialized.
+            let state = state.authorize(*nonce_authority).unwrap();
+            account.set_state(&state)
```

Comment:
Note a bit of new code here.

Reply:
It is part of removing `separate_domains` arg entirely from the code-base.

Reply:
This is now broken into 2 commits:
* 1st commit encapsulates AuthorizeNonceAccount implementation in the nonce module.
* 2nd commit removes the feature gate code.